src/widgets/models/CategoryModel.js

```diff
@@ -1,33 +1,44 @@
 import { executeSQL } from '../../api';
-import { filtersToSQL, viewportToSQL } from '../../api/FilterQueryBuilder';
+import { filtersToSQL } from '../../api/FilterQueryBuilder';
+import {groupValuesByColumn} from '../operations/grouping';
 
 export const getCategories = async (props) => {
-  const { data, credentials, column, operation, filters, viewport, opts } = props;
+  const { data, credentials, column, operation, filters, viewport, dataSource, viewportFilter, viewportFeatures, layerType, opts } = props;
 
   const operationColumn = props.operationColumn || column;
 
   if (Array.isArray(data)) {
     throw new Error('Array is not a valid type to get categories');
   }
 
-  let query =
-    (viewport && `SELECT * FROM (${data})  as q WHERE ${viewportToSQL(viewport)}`) ||
-    data;
+  // TODO: need to move this computation to a webworker
+  if (viewportFilter || layerType === 'bq') {
```

Comment:
That's not right. if a source is 'bq' and viewportFilter is not set we should throw an exception